### DISCUSSION TITLE
install/community/k8s/offline/

### DISCUSSION THREAD

**Discussion by giscus** [DaoCloud/DaoCloud-docs#1241] [Announcements]
# install/community/k8s/offline/

DaoCloud Enterprise 5.0 下一代容器化平台集大成者，引领云原生浪潮，推动数字化转型。

https://docs.daocloud.io/install/community/k8s/offline/

**Comment by wawa0210** [comment]
离线环境缺一个下载 `dce-installer` 二进制的步骤

**Comment by XilongPei** [comment]
标题"安装依赖项"，应该写完善些，比如这样写：
安装依赖项，在线或离线安装办法请点此链接。

**Comment by samzong** [comment]
缺少一个安装 dce5-installer 的步骤

**Comment by zhaoll-py** [comment]
7:25PM: Release "istio-ingressgateway" does not exist. Installing it now.
7:30PM: Error: context deadline exceeded
7:30PM: cleanup
7:30PM: [sched-info] Sched shutdown...
7:30PM: --------------------------------------
7:30PM: Unfortunately, the dce5 installation failed at step [8] ....
7:30PM: 
7:30PM: You can add -j8+ to your installation command to retry from last step.
7:30PM: --------------------------------------
[Error]:Failed to Install DCE5, check above error messages. shell execution err is exit status 1

**Comment by zhaoll-py** [comment]
在线安装和离线安装教程一样？

**Comment by zhaoll-py** [comment]
我就想知道你这个离线安装包下载下来有个屁用？所有镜像还是从网上下载，而且全部下载失败。跟在线安装没有任何区别。
安装文档写的稀里糊涂，简直最失败的产品。

**Reply by samzong** [reply]
Hi，麻烦将您的操作记录更详细的提供下，以便于分析问题；建议加入我们的社区群，我们有专业的支持人员。

感谢您的时间来体验DCE，对为您带来的问题感到抱歉。

希望和您取得联系，并解决此次遇到的问题。

**Reply by jzhupup** [reply]
在线跟离线的 clusterconfig 不一样呀？ 离线要定义一下 fullPackagePath: 离线包目录，如果定义了还是从网上下载我们需要排查下

**Reply by windsonsea** [reply]
Updated some content in PR #5201

**Comment by monday2k** [comment]
kpanda-clusterpedia-apiserver服务提示
authentication.go:73] "Unable to authenticate the request" err="[x509: subject with cn=front-proxy-client is not in the allowed list, verifying certificate

**Comment by wangyao980112** [comment]
为什么离线安装还要自己提供一个镜像库，这跟在线安装还有区别吗？

**Reply by jzhupup** [reply]
离线：离线安装下，不依赖外网环境，需要保证用户自己内网的网络互通性，安装所需的镜像需要导入到内网内的镜像仓库

在线，无需提供镜像仓库，所有镜像的拉取通过道客官方的镜像仓库获取，需要保证能访问公网

**Reply by wangyao980112** [reply]
好的，明白了

**Comment by zhudy517** [comment]
Back-off pulling image "192.168.***.***/docker.m.daocloud.io/istio/pilot:1.16.1"  

离线仓库pull不过来，同仓库下其他镜像有下载次数

**Comment by liuliu933** [comment]
10:22AM: [Error] Background job 106404 failed for sub folder multi-arch-images image pushing  , exit code 1 !, detail log in /var/lib/dce5/tmp/tmp.7pEk7Ex5wt/multi-arch-images.log
10:22AM:         Log detail for multi-arch-images:
10:22AM: time="2025-08-01T10:19:01+08:00" level=fatal msg="copying image 3/4 from manifest list: Uploading manifest failed, attempted the following formats: application/vnd.oci.image.manifest.v1+json(writing manifest: uploading manifest sha256:91c2c28a637a8166333cabd0e1acc9671218130933c2608659a69298d14c2d2a to 10.50.10.144:5000/release.daocloud.io/kpanda/kpanda-shell: manifest invalid), application/vnd.docker.distribution.manifest.v2+json(creating an updated image manifest: Unknown media type during manifest conversion: \"application/vnd.in-toto+json\"), application/vnd.docker.distribution.manifest.v1+prettyjws(creating an updated image manifest: Unknown media type during manifest conversion: \"application/vnd.in-toto+json\"), application/vnd.oci.image.index.v1+json(creating an updated image manifest: Unsupported conversion type: application/vnd.oci.image.index.v1+json), application/vnd.docker.distribution.manifest.list.v2+json(creating an updated image manifest: Unsupported conversion type: application/vnd.docker.distribution.manifest.list.v2+json), application/vnd.docker.distribution.manifest.v1+json(creating an updated image manifest: Unknown media type during manifest conversion: \"application/vnd.in-toto+json\")"
10:22AM: [Error] As above, common::skopeo_copy_imges_to_registry Failed ! target registry (10.50.10.144:5000)
10:22AM: cleanup

这个报错怎么回事，如何解决

**Reply by ErikJiang** [reply]
镜像中包含 application/vnd.in-toto+json 这种媒体类型（通常用于供应链安全签名），而目标 registry (10.50.10.144:5000) 不支持这种格式。

建议排查一下 registry 的版本，较旧的版本可能会存在不支持情况

**Reply by liuliu933** [reply]
谢谢，镜像的解决了

**Comment by liuliu933** [comment]
9:38AM: E0801 09:38:58.323231   74769 memcache.go:255] couldn't get resource list for custom.metrics.k8s.io/v1beta1: Got empty response for: custom.metrics.k8s.io/v1beta1
9:38AM: Error: rendered manifests contain a resource that already exists. Unable to continue with install: ValidatingWebhookConfiguration "istiod-default-validator" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "istio-base"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "istio-system"  未指定-c，直接安装报这个错误，是哪里还需要准备吗？

**Comment by gwhad3000** [comment]
4:42PM:  Installing Istiod ..
4:42PM: Successfully helm upgrade --install istiod /usr/local/soft/daolound/offline/infrastructure/istiod-1.22.3.tgz -n istio-system --version=1.22.3 -f /var/lib/dce5/tmp/tmp.ZusdcF8M9R -f /var/lib/dce5/tmp/values/tmp.i776YinjQh --wait --insecure-skip-tls-verify
4:42PM: deployment.apps/istiod patched (no change)
4:42PM: deployment.apps/istiod condition met
4:42PM:  Installing istio-ingressgateway..
4:42PM: [Error] Version comparison requires two non-empty arguments, got: '1.22.3' and ''
4:47PM: [Error] Failed to do helm. detail logs as below:
4:47PM: Error: UPGRADE FAILED: context deadline exceeded
4:47PM: + rm -f /tmp/tmp.pWodPax7XY
4:47PM: + exit 3
4:47PM: + cleanup
4:47PM: cleanup
4:47PM: + echo cleanup
4:47PM: + set +ex
4:47PM: [sched-info] Sched shutdown...
4:47PM: archive file from /var/lib/dce5/helm_upgrade_revisions.txt to /var/lib/dce5/helm_revisions/20260408_164728.txt successfully
4:47PM: --------------------------------------
4:47PM: Unfortunately, the DaoCloud community version installation failed at step [8] ....
4:47PM: 
4:47PM: You can add -j8+ to your installation command to retry from last step.
4:47PM: --------------------------------------

这个错误怎么处理啊？ 装了好多次每次都卡在这里

**Reply by gwhad3000** [reply]
有没有人帮忙看下什么问题？

**Reply by ErikJiang** [reply]
用如下命令验证一下当前组件的版本信息是否可以正常获取到：

``` bash 
helm history <release> -n <ns> -o yaml | yq '.[] | select(.status == "deployed") | [.app_version] | .[(length - 1)]'
```
若获取不到，请检查一下该组件部署状态，保障运行健康后再重试

**Reply by gwhad3000** [reply]
就是部署不成功啊，用这个语句检查了，helm history istio-ingressgateway -n istio-system -o yaml 都是description: 'Upgrade "istio-ingressgateway" failed: context deadline exceeded' 
既然是离线安装，为什么还会有这种错误？
